multinet/db/models/table.py

```diff
@@ -72,6 +81,32 @@ def headers(self) -> List[str]:
 
         return keys
 
+    def get_metadata(self) -> Optional[ArangoEntityDocument]:
+        """Retrieve metadata for this table, if it exists."""
+        try:
+            doc = next(self.metadata_collection.find({"item_id": self.name}, limit=1))
+        except StopIteration:
+            return None
+
+        return ArangoEntityDocument(**doc)
+
+    def set_metadata(self, raw_data: Dict) -> ArangoEntityDocument:
+        """Set metadata for this table."""
+        try:
+            data = TableMetadata(**raw_data)
+        except PydanticValidationError:
+            raise InvalidMetadata(raw_data)
+
+        existing = self.get_metadata()
+        if existing is not None:
+            existing.table = data
+            doc_meta = self.metadata_collection.update(existing.dict())
+        else:
+            entity = EntityMetadata(item_id=self.name, table=data, graph=None)
+            doc_meta = self.metadata_collection.insert(entity.dict())
```

Comment:
This is not at all a big deal, but it feels like you could condense this code quite a bit just by letting the `EntityMetadata` initializer not have to take the `table` and `graph` arguments and set them both to `None` initially, then explicitly set the one that you do want. Then, line 100 could be `entity = self.get_metadata() or EntityMetadata(item_id=self.name)`, followed immediately by `entity.table = data` and then `doc_meta = self.metadata_collection.upsert(entity.dict())`. The total code block would look like this:

```suggestion
        entity = self.get_metadata() or EntityMetadata(item_id=self.name)
        entity.table = data
        doc_meta = self.metadata_collection.insert(entity.dict(), overwrite=True)
```

We'd have to make sure that the `overwrite = True` kwarg makes `.insert()` behave like an upsert. And this may not be worth doing at all, but I just like how the shorter code reads.

This is also making me think the whole `table`/`graph` metadata section split is an antipattern but we can fix that if it turns out to be a problem.

Reply:
It [looks](https://python-driver-for-arangodb.readthedocs.io/en/master/specs.html#arango.collection.StandardCollection.insert) like `overwrite` combined with `overwrite_mode` would give us the `upsert` behavior we desire. I think your code does read better, so thanks for that. I would commit this, but since I'll be changing the behavior for missing metadata as discussed above, this code will change slightly. I'll make this change after updating the behavior for `get_metadata`.

Reply:
Actually, I was looking at newer docs than what we're running. It seems if `overwrite=True`, the existing doc will be replaced with the doc we specify, which I think is the correct behavior.

Reply:
6118164